exporters/prometheus/exporter.go

```diff
@@ -122,6 +128,12 @@ type metricData struct {
 
 func getMetricData(metrics metricdata.ResourceMetrics) []*metricData {
 	allMetrics := make([]*metricData, 0)
+
+	targetInfo := createInfoMetricData(targetInfoMetricName, targetInfoDescription, metrics.Resource)
+	if targetInfo != nil {
+		allMetrics = append(allMetrics, targetInfo)
+	}
```

Comment:
This should only be computed once given we implicitly expect there to only ever be one target info metric generated.

Reply:
Couldn't an exporter be shared between different `MeterProvider`s or is that something that we shouldn't support at all?

Reply:
[Readers are only allowed to be registered once](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.14.0/specification/metrics/sdk.md#metricreader):

> The SDK MUST NOT allow a MetricReader instance to be registered on more than one MeterProvider instance.

We guard for this: https://github.com/open-telemetry/opentelemetry-go/blob/b6a22abab751307b4851b3cc51728fa557f907bb/sdk/metric/manual_reader.go#L54-L58

The only way multiple resources would appear is if the exporter shared a registerer.